src/zenml/integrations/azure/orchestrators/azureml_orchestrator.py

```diff
@@ -0,0 +1,545 @@
+                    start_time = schedule.start_time.isoformat()
+
+                end_time = None
+                if schedule.end_time is not None:
+                    end_time = schedule.end_time.isoformat()
+
+                if schedule.cron_expression:
+                    # If we are working with a cron expression
+                    schedule_trigger = CronTrigger(
+                        expression=schedule.cron_expression,
+                        start_time=start_time,
+                        end_time=end_time,
+                        time_zone=TimeZone.UTC,
+                    )
+
+                elif schedule.interval_second:
+                    # If we are working with intervals
+                    interval = schedule.interval_second.total_seconds()
+
+                    if interval % 60 != 0:
+                        logger.warning(
+                            "The ZenML AzureML orchestrator only works with "
+                            "time intervals defined over minutes. Will "
+                            f"use a schedule over {int(interval // 60)}."
+                        )
+
+                    if interval < 60:
+                        raise RuntimeError(
+                            "Can not create a schedule with an interval less "
+                            "than 60 secs."
+                        )
+
+                    frequency = "minute"
+                    interval = int(interval // 60)
+
+                    schedule_trigger = RecurrenceTrigger(
+                        frequency=frequency,
+                        interval=interval,
+                        start_time=start_time,
+                        end_time=end_time,
+                        time_zone=TimeZone.UTC,
+                    )
+
+                if schedule_trigger:
+                    # Create and execute the job schedule
+                    job_schedule = JobSchedule(
+                        name=run_name,
+                        trigger=schedule_trigger,
+                        create_job=pipeline_job,
+                    )
+                    ml_client.schedules.begin_create_or_update(
+                        job_schedule
+                    ).result()
+                    logger.info(
+                        f"Scheduled pipeline '{run_name}' with recurrence "
+                        "or cron expression."
+                    )
+                else:
+                    raise RuntimeError(
+                        "No valid scheduling configuration found for "
+                        f"pipeline '{run_name}'."
+                    )
+
+            except (HttpResponseError, ResourceExistsError) as e:
+                raise RuntimeError(
+                    "Failed to create schedule for the pipeline "
+                    f"'{run_name}': {str(e)}"
+                )
+
+        else:
+            ml_client.jobs.create_or_update(pipeline_job)
+            logger.info(f"Pipeline {run_name} has been started.")
```

Comment:
It would've been nice to have sync/async mode support for this orchestrator, as we have for most of the other ones :). Maybe create a ticket to implement it later ?

Reply:
Sounds like a good idea. Will create the ticket.